databases/mongoose.md

```diff
@@ -0,0 +1,275 @@
+                }
+            });
+    ```
+    It's that easy to find information from the database.   
+
+
+7. Deleting all information on the database: ```deleteMany() method```
+
+    ```javascript
+    Dog.deleteMany({}, (err,dogs) => {
+        if (err) {
+            console.log(err);
+         }
+        else {
+            //should return null since all dogs are deleted
+            console.log(dogs)
+        }
+    });
+    ```
+
+8. Deleting a particular object by finding it by property:```deleteOne() method```
+
+    ```javascript
+    Dog.deleteOne({name:"Siri"},  (err, dog)=> {
+    if (err) {
+        console.log(err);
+    }
+    else {
+        //deletes the dog information with name Siri
+        console.log(dog);
+    }
+    });
+    ```
+    It's that easy to delete a data from the database.
+
+9. Finding a data and updating it: ```findByIdAndUpdate() method```
+
+    ```javascript
+    const idToUpdate="5db5624e582bfd30145a2e09";
+    const updatedInfo={
+        name:"Loki",
+        age:13,
+        breed:"Belgium Shepherd",
+        favouriteFood:"Biscuits"
+    
+    }
+     Dog.findByIdAndUpdate(idToUpdate,updatedInfo, (err, updatedDog)=> {
+        if (err) {
+            console.log(err);
+        } else {
+            //print the updated Dog
+            console.log(updatedDog);
+
+       }
+    });
+    
+    ```
+    It's that easy to update information using mongoose using it's built in methods.
+
+It's important to observe a pattern in all of the query methods which follows a basic principle of calling a method which usually takes in arguments:  
+1st argument: Usually used to provide information to the database (used in findOne,create,deleteOne methods).  
+2nd argument: Usually used in providing an updated information on the database for an already existing data.  
+3rd argument: A callback function which tells you whether a error has occured or not in performing an operation .  
+
+
+
+### Independent Practice
+
+1. Read through these mongoose docs (especially Queries Section) from official website and go through all the mentioned queries and think about how you can use them. 
+    Check out the resources [here](https://mongoosejs.com/docs/queries.html).
+
+2. Try to create your own RESTful Api that uses mongoose to perform all DB operations.
+    Check out a tutorial [here](https://www.youtube.com/watch?v=vjf774RKrLc).
+
+### Challenge
+How cool would it be if we can store our data to the cloud instead of our local computer. This would be a practical way of deploying your application as you want your data to be stored on any Cloud Service.  
+Why not try to replace the API you build to a cloud based API by using [MongoDB Atlas](https://www.mongodb.com/cloud/atlas).  
+Go through the documentation and find out the ways you can do this. Share the application with your frinds and let them see your work.  
+
+Hint: Build a small Blog Post application where you can add,read,edit or update a blog.  
```

Comment:
```suggestion
```
They will be using their new Mongoose skills for a project in a few days; don't worry about assigning them one.